File: src/domain/entities/messaging/DeliveryResult.ts
import { SessionReference } from '../operational/SessionReference.js';

export enum DeliveryStatus {
  Sent = 'sent',
  Failed = 'failed',
  Blocked = 'blocked',
}

export class DeliveryResult {
  constructor(
    public readonly commandId: string,
    public readonly session: SessionReference,
    public readonly recipientId: string,
    public readonly status: DeliveryStatus,
    public readonly timestamp: string,
    public readonly providerMessageId?: string,
    public readonly reason?: string,
  ) {}
}

File: sdks/java/src/main/java/com/digows/whatsappgateway/messaging/DeliveryResult.java
package com.digows.whatsappgateway.messaging;

import com.digows.whatsappgateway.operational.SessionReference;

public record DeliveryResult(
  String commandId,
  SessionReference session,
  String recipientId,
  DeliveryStatus status,
  String timestamp,
  String providerMessageId,
  String reason
)
{
}
